text/0000-fabric-gateway.md

```diff
@@ -0,0 +1,233 @@
+
+Trusting clients can call the SDK's submitTransaction function to endorse/sign/submit in a single line of code for convenience. Users of the existing high level programming model are already comfortable with this procedure. 
+
+## Authentication and Authorization
+
+The Gateway itself will not add any message level authentication mechanism other than that already provided by the peers.  As a proxy, authentication and authorization is handled by peer/orderer nodes exactly as if the client was connecting to them directly.
+
+Connection of the client applications to the Gateway can be secured by mutual TLS if required.
+
+The standalone Gateway does need to have an identity that allows it to connect to peers as a client in its own right to:
+- Do network discovery to obtain network topology and endorsement plans.
+- Listen to events to allow it to check for commit disposition of submitted transactions that it has routed to the ordering service on behalf of a client.
+
+## Gateway SDKs
+
+A set of SDKs will be created to allow client applications to interact with the Fabric Gateway.  The APIs exposed by these SDKs will be, where possible, the same as the current high-level 'gateway' SDKs.
+
+The following classes/structures will be available as part of the gateway programming model:
+- Gateway
+- Network
+- Contract
+- Transaction
+
+The existing `submitTransaction()` and `evaluateTransaction()` API methods will be implemented using the Gateway gRPC services described above.
+
+Based on user feedback of the existing gateway SDKs, the following additions are planned to be supported by this implementation:
+- Support for 'offline signing', allowing clients to write REST servers that don't directly have access to users' private keys.
+- Support for 'asynchronous submit', allowing client applications to continue processing while the transaction commit proceeds in the background. For Go this will be implemented by returning a channel; for Node by returning a Promise; and for Java by returning a Future.
+
+SDKs will be created for the following languages:
+- Node (Typescript/Javascript)
+- Go
+- Java
+
+It is proposed that these SDKs will be maintained in the same GitHub repository as the gateway itself (`hyperledger/fabric-gateway`) to ensure that they all stay up to date with each other and with the core gateway.  This will be enforced in the CI pipeline by running the end-to-end scenario tests across all SDKs.
+
+Publication of SDK releases will be as follows:
+- Node SDK published to NPM.
+- Java SDK published to Maven Central.
+- Go SDK published to separate GitHub repo (e.g. fabric-gateway-go) and tagged.
```

Comment:
Are we still planning to have another github repo for Go SDK release like hyperledger/fabric-gateway-go?
I see `module github.com/hyperledger/fabric-gateway` is also enough if not conflicting with other language SDK.

Reply:
The source repository (hyperledger/fabric-gateway) is the Go module location. No plans anymore to create a mirror repository to host only the published Go code.